src/ray/protobuf/gcs.proto

```diff
@@ -690,4 +665,32 @@ message PlacementGroupTableData {
   // effectively when the placement group has been "scheduled".
   int64 placement_group_final_bundle_placement_timestamp_ms = 15;
 }
+
+message JobTableData {
+  // The job ID.
+  bytes job_id = 1;
+  // Whether it's dead.
+  bool is_dead = 2;
+  // The UNIX timestamp corresponding to this event (job added or removed).
+  int64 timestamp = 3;
+  // IP address of the driver that started this job.
+  // Deprecated
+  string driver_ip_address = 4;
+  // Process ID of the driver running this job.
+  int64 driver_pid = 5;
+  // The config of this job.
+  JobConfig config = 6;
+  // The timestamp the job was started at.
+  uint64 start_time = 7;
+  // The timestamp the job was ended at.
+  uint64 end_time = 8;
+  // The entrypoint of the job.
+  string entrypoint = 9;
+  // Whether this job has running tasks.
+  bool is_running_tasks = 10;
+  // Address of the driver that started this job.
+  Address driver_address = 11;
+  // The optional JobInfo from the Ray Job API.
+  optional JobsAPIInfo job_info = 12;
```

Comment:
Is this compatible with the what's before the broken PR and status quo? I thought field numbers are not supposed to be changed to be backward compatible? 

Which old version are we making this backward compatible (since we could not be compatible to both)?

Reply:
It is to ray 2.5.1. Actually I am also not sure if the order should be reserved even for optional. Should we?

Reply:
It is for 2.5.1. So even optoinal, the ordering should be reserved?

Reply:
Yeah, i believe so? 
> When you no longer need a non-required field and all references have been deleted from client code, you may delete the field definition from the message. However, you must [reserve the deleted field number](https://protobuf.dev/programming-guides/proto3/#fieldreserved). If you do not reserve the field number, it is possible for a developer to reuse that number in the future.

> Don’t change the field numbers for any existing fields. “Changing” the field number is equivalent to deleting the field and adding a new field with the same type. If you want to renumber a field, see the instructions for [deleting a field](https://protobuf.dev/programming-guides/proto3/#deleting).

https://protobuf.dev/programming-guides/proto3/#deleting

Reply:
Thanks for catching that @rickyyx , I think you're right. So concretely, we're saying `driver_address` should be after `job_info`. I didn't find any documentation saying that `optional` fields need to be at the end.

Reply:
So maybe we choose which version we are supporting (since it has broken one of the two) - and keep the numbering consistent with that version?

Reply:
Fixed db2fc81